### PR TITLE
Don't prefetch v2 checkpoints twice, check first if one is already ongoing.

### PR DESCRIPTION
**Pull request by nick4598**
imodel-native: https://github.com/iTwin/imodel-native/tree/nick/prefetch-client-counts

### CODE REVIEW

**Comment by khanaffan** [comment]
@nick4598 I do not see any unit test with changes.

**Comment by nick4598** [comment]
> @nick4598 I do not see any unit test with changes.

Thanks added a test

File: core/backend/src/CheckpointManager.ts [hunk 1]
```diff
@@ -225,7 +225,7 @@ export class V2CheckpointManager {
           // eslint-disable-next-line @typescript-eslint/no-floating-promises
           logPrefetch(CloudSqlite.startCloudPrefetch(container, dbName, { minRequests, nRequests: maxRequests, timeout }));
         } else {
-          Logger.logInfo(loggerCategory, `Skipping prefetch due to size limits.`, { maxBlocks, totalBlocksInDb: dbStats?.totalBlocks, v2props });
+          Logger.logInfo(loggerCategory, `Skipping prefetch due to size limits or ongoing prefetch.`, { maxBlocks, numPrefetches: dbStats?.nPrefetch, totalBlocksInDb: dbStats?.totalBlocks, v2props });
```
Comment (kabentley) [inline core/backend/src/CheckpointManager.ts:228]: probably helpful to say whether we're skipping due to an active prefetch already in progress or it being disabled.

**Comment by kabentley** [comment]
I frankly don't understand why this comes up. Do we really start more than one backend on the same node for a single iModel?

File: core/backend/src/CheckpointManager.ts (hunk 1 above)
Comment (nick4598) [reply core/backend/src/CheckpointManager.ts:228]: Yeah I thought about having separate log messages for each case, but as it is currently we can deduce which case we are in by looking at what was logged. The maxBlocks limit, totalBlocks in the database, and the number of active prefetches. Between those 3, we can figure out the case. Do you still prefer a separate log message?

**Comment by nick4598** [comment]
> I frankly don't understand why this comes up. Do we really start more than one backend on the same node for a single iModel?

This is fairly new, DR has recently started pointing their frontend traffic for iModelReadRPCInterfaces, PresntationRPCinterface etc to GPB. This would mean two backends for the same iModel. There is some preference in k8s to put pods with the same iModelid together, but its not guaranteed. In the case that they aren't on the same node, DR would also trigger a prefetch on its own.

File: core/backend/src/CheckpointManager.ts (hunk 1 above)
Comment (kabentley) [reply core/backend/src/CheckpointManager.ts:228]: as long as you have something you can search for in the logs to find when multiple prefetches would have happened, that's fine

**Review by kabentley — APPROVED** [review]

**Comment by nick4598** [comment]
@mergifyio backport release/4.1.x

**Pull request merged**